pkg/prometheus/promcfg.go

```diff
@@ -1004,6 +1004,15 @@ func (cg *ConfigGenerator) BuildCommonPrometheusArgs() []monitoringv1.Argument {
 		}
 	}
 
+	// Turn on the OTLP receiver endpoint automatically if/when the OTLP config isn't empty.
+	if (cpf.EnableOTLPReceiver != nil && *cpf.EnableOTLPReceiver) || cpf.OTLP != nil {
```

Comment:
(nit) for consistency, the enabled flag should prevail if explicitly set to false.
```suggestion
	if (cpf.EnableOTLPReceiver != nil && *cpf.EnableOTLPReceiver) || (cpf.EnableOTLPReceiver == nil && cpf.OTLP != nil) {
```